### PR TITLE
Bootstrap study

### PR DESCRIPTION
**Pull request by jackbsteinberg**

### CODE REVIEW

File: study-group/Toast-Implementation-Studies.md [hunk 1]
```diff
@@ -407,9 +407,108 @@ The library allows showing multiple toasts, and gives the developer specific con
...
+      <span aria-hidden="true">&times;</span>
+    </button>
+  </div>
+  <div class="toast-body">
+    Hello, world! This is a Bootstrap Toast!
+  </div>
+</div>
+```
+
+Bootstrap uses classes to denote special elements of the components it constructs, like the classes `toast`, `toast-header`, and `toast-body`.
```
Comment (fergald) [inline study-group/Toast-Implementation-Studies.md]: this file does not seem to be following semantic line-breaking.

File: study-group/Toast-Implementation-Studies.md [hunk 2]
```diff
@@ -407,9 +407,108 @@ The library allows showing multiple toasts, and gives the developer specific con
...
+The example uses many `aria` attributes, indicating a deliberate effort for accessibility.
+Multiple Toasts can be wrarpped together in a `div` to stack intuitively.
```
Comment (fergald) [inline study-group/Toast-Implementation-Studies.md]:
```suggestion
Multiple Toasts can be wrapped together in a `div` to stack intuitively.
```

Comment (fergald) [inline study-group/Toast-Implementation-Studies.md]: I'm not sure what "stack intuitively" means. Might be good to include images and now that I think of it, images in some of the other studies could be helpful too (I don't have specific examples but maybe having a snapshot of any interesting examples would be good. Nobody is likely to actually make one of these examples run to see what it looks like.

Comment (jackbsteinberg) [reply study-group/Toast-Implementation-Studies.md]: That's a good idea, something I had discussed with Domenic was creating a file of informative snapshots to go along with the study and link relevant ones to sections in this doc. Do you think I should do that, or just add in a few to this doc?

File: study-group/Toast-Implementation-Studies.md (hunk 1 above)
Comment (jackbsteinberg) [reply study-group/Toast-Implementation-Studies.md]: My bad, it's still been a bit strange to get used to, I'll go through and change that everywhere I can find

File: study-group/Toast-Implementation-Studies.md (hunk 2 above)
Comment (jackbsteinberg) [reply study-group/Toast-Implementation-Studies.md]: @fergald I'm going to make a branch once the content is all pushed in to clean it up, by editing and adding some relevant images.

Comment (fergald) [reply study-group/Toast-Implementation-Studies.md]: Yeah, it's not urgent but I think surveys of UI components will benefit from images (but this being mostly an API survey, not so much, mostly for demonstrating results of non-obvious APIs).

File: study-group/Toast-Implementation-Studies.md [hunk 3]
```diff
@@ -10,8 +10,8 @@ The libraries studied are:
 - [Material-UI Snackbar](#material-ui-snackbar)
 - [Android `Toast`](#android-toast)
 - [react-toastify](#react-toastify)
-- [ngx-toastr](#ngx-toastr)
-- [Bootstrap](#bootstrap)
+- [ngx-toastr](#ngx-toast)
```
Comment (domenic) [inline study-group/Toast-Implementation-Studies.md]: Was this change intended?

File: study-group/Toast-Implementation-Studies.md [hunk 4]
```diff
@@ -407,9 +413,109 @@ The library allows showing multiple toasts, and gives the developer specific con
...
+      <div class="toast-body">
+        Heads up, Toasts will stack automatically
+      </div>
+    </div>
+  </div>
+</div>
+```
+
+Using native styling with the `position: absolute;` attribute on, the Toasts can be intuitively positioned on the screen.
+Note that the order the Toasts display is determined by their DOM order, not the order in which they're displayed.
```
Comment (domenic) [inline study-group/Toast-Implementation-Studies.md]: This sentence sounds a little funny, although I know what you mean. Could it be rephrased?

File: study-group/Toast-Implementation-Studies.md [hunk 5]
```diff
@@ -407,9 +413,109 @@ The library allows showing multiple toasts, and gives the developer specific con
...
+#### Pure HTML View
+Bootstrap Toasts are purely HTML elements highly styled with CSS, and are easily customizable.
+
+#### Events
+The library offers events for `show`, `shown`, `hide`, and `hidden`, to represent the moments before and after the Toast is shown / hidden.
+
+#### Accessibility
+The documentation is explicit about suggesting wrapping the toasts in `aria-live`, to announce the content via screen reader without moving focus.
+It also recommends including `aria-atomic="true"` to ensure the Toast is read as a single atomic element, 
+and live changes don't disrupt the screen-reader user.
```
Comment (domenic) [inline study-group/Toast-Implementation-Studies.md:506]: This is great info!

File: study-group/Toast-Implementation-Studies.md [hunk 6]
```diff
@@ -407,9 +413,109 @@ The library allows showing multiple toasts, and gives the developer specific con
...
+Bootstrap components that are meant to be overlayed use an arbitrary z-index value, in the range of ~1000.
+
+### Takeaways
+
+- Low JavaScript configuration, high HTML customization.
+- Uses the same `.toast(options)` method for `show`, `hide`, and `dispose`, by passing in any of those strings as `options`.
+- The developer needs to be cognizant of DOM order for multiple toasts.
+  -  Calling `show` out order would still display Toasts in DOM order, a potentially unexpected behavior.
+- Requires `util.js` if building the JavaScript from source.
+- When disabling `autohide` the developer must provide a close button, it will not be added by default.
```
Comment (domenic) [inline study-group/Toast-Implementation-Studies.md]:
```suggestion
- When disabling `autohide` the developer must provide a close button; it will not be added by default.
```

**Review by domenic — APPROVED** [review]
LGTM with nits

File: study-group/Toast-Implementation-Studies.md (hunk 3 above)
Comment (jackbsteinberg) [reply study-group/Toast-Implementation-Studies.md]: Probably not, I think that came from some of the merge shenanigans
```suggestion
- [ngx-toastr](#ngx-toastr)
```

File: study-group/Toast-Implementation-Studies.md (hunk 5 above)
Comment (jackbsteinberg) [reply study-group/Toast-Implementation-Studies.md:506]: Thanks 👍

File: study-group/Toast-Implementation-Studies.md (hunk 4 above)
Comment (jackbsteinberg) [reply study-group/Toast-Implementation-Studies.md]: Yeah I'll try and rephrase it

Comment (jackbsteinberg) [reply study-group/Toast-Implementation-Studies.md]:
```suggestion
Note that the order the Toasts display in is determined by their DOM order, _not_ the order in which they are triggered.
```
